core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java

```diff
@@ -159,6 +159,8 @@ protected void initChannel(Channel ch) throws Exception {
 
     private void configurePipeline(ChannelPipeline p, Set<SessionProtocol> protocols,
                                    @Nullable ProxiedAddresses proxiedAddresses) {
+        config.channelPipelineCustomizer().accept(p);
```

Comment:
I've tried, but `config.childChannelPipelineCustomizer().accept(p)` seems not to have an effect on pipeline in case of http.
(It's OK in case of https)

Apparently, it occurs if `config.childChannelPipelineCustomizer().accept(p)` is processed after [`p.addLast(new Http2PrefaceOrHttpHandler(responseEncoder));`](https://github.com/line/armeria/blob/0139a1ead05dbc09c7027b7ca01596f2afc94e15/core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java#L217).
I'm not sure why this happens. Do you have any ideas?

Also, I've added the ParameterizedTests of http and https patterns in `ServerChildChannelPipelineCustomizerTest` in order to verify each Http protocol patterns.

Reply:
I've moved this line to the end of [`initChannel()`](https://github.com/line/armeria/blob/0139a1ead05dbc09c7027b7ca01596f2afc94e15/core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java#L157)
Accordingly, I corrected like we add a `ChannelHandler` in the beginning of the pipeline instead of the last in `ServerChildChannelPipelineCustomizerTest`.

Reply:
Do you still have the problem you mentioned in the second comment in this thread (i.e. `accept()` has no effect)?

Reply:
Yes I do.
If I move `config.channelPipelineCustomizer().accept(p);` to the end of [`initChannel()`](https://github.com/line/armeria/blob/0139a1ead05dbc09c7027b7ca01596f2afc94e15/core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java#L157), there success and fail test patterns like the following:

Server: This is a sccess test pattern. Handler is added to the pipeline by `addFirst`.
```java
@RegisterExtension
static ServerExtension server = new ServerExtension() {
    @Override
    protected void configure(ServerBuilder sb) throws Exception {
        sb.http(0);
        sb.https(0);
        sb.tlsSelfSigned();
        sb.childChannelPipelineCustomizer(
                pipeline ->  pipeline.addFirst(new ChannelTrafficShapingHandler(1024, 0)));
        sb.service("/", new AbstractHttpService() {
            @Override
            protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
                return HttpResponse.of(
                        ResponseHeaders.of(HttpStatus.OK, "header1", Strings.repeat("a", 2048)));
            }
        });
    }
};
```

Server: This is a fail test pattern. Handler is added to the pipeline by `addLast`.

```java
...
sb.childChannelPipelineCustomizer(
                pipeline ->  pipeline.addLast(new ChannelTrafficShapingHandler(1024, 0)));
...
```

Again, I debugged and clarified the cause.
If a handler specified above codes is added after [`p.addLast(new Http2PrefaceOrHttpHandler(responseEncoder));`](https://github.com/line/armeria/blob/0139a1ead05dbc09c7027b7ca01596f2afc94e15/core/src/main/java/com/linecorp/armeria/server/HttpServerPipelineConfigurator.java#L217), it seems to have no effect in case of http. (It's OK in case of https)